pkg/sync/kubernetes/kubernetes_sync.go

```diff
@@ -91,101 +130,47 @@ func (k *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
 	}
 }
 
+// fetch attempts to retrieve the latest feature flag configurations
 func (k *Sync) fetch(ctx context.Context) (string, error) {
-	if k.URI == "" {
-		k.Logger.Error("no target feature flag configuration set")
-		return "{}", nil
-	}
-
-	ns, name, err := parseURI(k.URI)
+	// first check the store - avoid overloading API
+	object, exist, err := k.informer.GetStore().GetByKey(k.URI)
```

Comment:
```suggestion
	rawCfg, exist, err := k.informer.GetStore().GetByKey(k.URI)
```

It doesn't have to be this name, but I think implying this is unparsed JSON or a raw payload would be clearer.

Reply:
resolved with the latest commit